### PR TITLE
sql server module in VNET

### PR DESCRIPTION
**Pull request by kristianschneider**
This PR extends the sql server module to ensure it works within a VNET
This resolves https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/the-outlaws/97

### CODE REVIEW

File: azure/sql-server/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ This module creates the following resources.
 
 ## Prerequisites
 
-- Terraform version 1.0.6+
+- Terraform version 1.1.12+
```
Comment (djorgensendk) [inline azure/sql-server/README.md:17]: 1) Just to understand, why is this change needed?
2) I can't find this version on https://github.com/hashicorp/terraform/releases - latest version is 1.1.3 released about a week ago.

File: azure/sql-server/README.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ This module creates the following resources.
 | `sql_version` | `string` | | **Required** | The version for the new server. Valid values are: 2.0 (for v11 server) and 12.0 (for v12 server). |
 | `administrator_login` | `string` | | **Required** | The administrator login name for the new server. Changing this forces a new resource to be created. |
 | `administrator_login_password` | `string` | | **Required** | The password associated with the administrator_login user. Needs to comply with Azure's [Password Policy](https://msdn.microsoft.com/library/ms161959.aspx) |
+| `private_endpoint_subnet_id` | `string` | | **Required**  The terraform id of the private endpoint subnet
```
Comment (djorgensendk) [inline azure/sql-server/README.md:33]: I guess it the ID of the resource deployed in Azure?
```suggestion
| `private_endpoint_subnet_id` | `string` | | **Required**  The ID of the private endpoint subnet
```

File: azure/sql-server/main.tf [hunk 3]
```diff
@@ -13,18 +13,19 @@
...
 locals {
   module_tags = {
-    "ModuleVersion" = "5.1.0"
+    "ModuleVersion" = "6.0.0"
     "ModuleId"      = "azure-sql-server"
   }
 }
 
-resource "azurerm_sql_server" "this" {
+resource "azurerm_mssql_server" "this" {
```
Comment (djorgensendk) [inline azure/sql-server/main.tf:21]: Will this change cause the SQL servers we currently have deployed in all environments to be deleted and afterwards new servers to be created? In this case all domains must be redeployed to seed them with new data

File: azure/sql-server/main.tf [hunk 4]
```diff
@@ -40,12 +41,23 @@ resource "azurerm_sql_server" "this" {
   }
 }
 
-resource "azurerm_sql_firewall_rule" "this" {
-  count               = length(var.firewall_rules)
-
-  name                = "${lower(try(var.firewall_rules[count.index].start_ip_address, null))}-${lower(var.name)}-${lower(var.project_name)}-${lower(var.environment_short)}-${var.environment_instance}"
+resource "azurerm_private_endpoint" "this" {
+   name                = "pe${lower(var.name)}${lower(var.project_name)}${lower(var.environment_short)}${lower(var.environment_instance)}"
```
Comment (djorgensendk) [inline azure/sql-server/main.tf:45]:
```suggestion
   name                = "pe-${lower(var.name)}${lower(var.project_name)}${lower(var.environment_short)}${lower(var.environment_instance)}"
```

File: azure/sql-server/main.tf [hunk 5]
```diff
@@ -40,12 +41,23 @@ resource "azurerm_sql_server" "this" {
...
+   location            = var.location
+   resource_group_name = var.resource_group_name
+   subnet_id           = var.private_endpoint_subnet_id
+   private_service_connection {
+     name                           = "psc${lower(var.name)}${lower(var.project_name)}${lower(var.environment_short)}${lower(var.environment_instance)}"
```
Comment (djorgensendk) [inline azure/sql-server/main.tf:50]:
```suggestion
     name                           = "psc-${lower(var.name)}${lower(var.project_name)}${lower(var.environment_short)}${lower(var.environment_instance)}"
```

File: azure/sql-server/main.tf (hunk 3 above)
Comment (kristianschneider) [reply azure/sql-server/main.tf:21]: azurerm_sql_server is depricated, So we need to move to this regardless. Question is how we handle redployments

Comment (djorgensendk) [reply azure/sql-server/main.tf:21]: of course we shouldn't use a deprecated resource. Great that you found this before we go into the actor test in a few weeks time.

File: azure/sql-server/README.md [hunk 6]
```diff
@@ -30,6 +30,9 @@ This module creates the following resources.
 | `sql_version` | `string` | | **Required** | The version for the new server. Valid values are: 2.0 (for v11 server) and 12.0 (for v12 server). |
 | `administrator_login` | `string` | | **Required** | The administrator login name for the new server. Changing this forces a new resource to be created. |
 | `administrator_login_password` | `string` | | **Required** | The password associated with the administrator_login user. Needs to comply with Azure's [Password Policy](https://msdn.microsoft.com/library/ms161959.aspx) |
+| `private_endpoint_subnet_id` | `string` | | **Required**  The ID of the private endpoint subnet
```
Comment (Renetnielsen) [inline azure/sql-server/README.md:33]: Move | as other PR

File: azure/sql-server/README.md [hunk 7]
```diff
@@ -30,6 +30,9 @@ This module creates the following resources.
...
+| `private_dns_zone_name` | `string` | | **Required**  The name of the private dns zone
```
Comment (Renetnielsen) [inline azure/sql-server/README.md:34]: Move | as other PR

**Review by Renetnielsen — APPROVED** [review]

**Review by djorgensendk — APPROVED** [review]

**Pull request merged**